espnet/asr/asr_mix_utils.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+import copy
+import logging
+# matplotlib related
+import os
+
+from chainer.training import extension
+
+# io related
+import kaldi_io_py
+import matplotlib
+import numpy as np
+
+'''
+reuse modules in asr_utils:
+    CompareValueTrigger, restore_snapshot, _restore_snapshot, adadelta_eps_decay,
+    _adadelta_eps_decay, torch_snapshot, _torch_snapshot_object, AttributeDict,
+    get_model_conf, chainer_load, torch_save, torch_load, torch_resume
+'''
+from espnet.asr.asr_utils import parse_hypothesis
+
+
+matplotlib.use('Agg')
+
+
+# * -------------------- training iterator related -------------------- *
+def make_batchset(data, batch_size, max_length_in, max_length_out,
```

Comment:
I’m not sure which part is modified.
If functions are the same, we do not add them here.
If there are some modifications, it is OK.

Reply:
The variable `olen` is different.